controller/controlplane/controller_reconciler_utils.go

```diff
@@ -348,12 +356,306 @@ func (r *Reconciler) ensureRoles(
 		return op.Updated, nil // requeue will be triggered by the creation or update of the owned object
 	}
 
+	// If watchNamespaces is not empty then we need to generate roles and role bindings for each namespace.
+	if len(generatedRoles) > 0 {
+		log.Trace(logger, "ensuring Roles for ControlPlane Deployment exist")
+		createdOrUpdated, controlplaneRoles, err := r.ensureRoles(ctx, cp, generatedRoles)
+		if err != nil {
+			return op.Noop, err
+		}
+		if createdOrUpdated {
+			log.Debug(logger, "Roles created/updated")
+			return op.Updated, nil // requeue will be triggered by the creation or update of Roles.
+		}
+
+		res, err := r.ensureRoleBindings(ctx, cp, controlplaneServiceAccount, controlplaneRoles)
+		if err != nil {
+			return op.Noop, err
+		} else if res != op.Noop {
```

Comment:
```suggestion
		}
		if res != op.Noop {
```
to be consitent with what is above and below